### PR TITLE
[PREGEL] Make pregel js tests more stable

### PR DESCRIPTION
**Pull request by jvolmer**
Use a function everywhere where a test waits for pregel to finsh successfully. This makes sure that all tests check against the correct pregel status and give helpful error message if the pregel run did not finish successfully.

### CODE REVIEW

File: tests/js/common/shell/shell-pregel-components.js [hunk 1]
```diff
@@ -200,29 +200,20 @@ function componentsTestSuite() {
...
+
+            assertEqual(stats.vertexCount, numComponents * n, stats);
+            assertEqual(stats.edgeCount, numComponents * (m + n), stats);
+
+            let c = db[vColl].all();
+            let mySet = new Set();
+            while (c.hasNext()) {
+                let doc = c.next();
+                assertTrue(doc.result !== undefined, doc);
+                mySet.add(doc.result);
```
Comment (markuspf) [inline tests/js/common/shell/shell-pregel-components.js:213]: I see this code repeated a few times; maybe factor it into separate function?

File: tests/js/common/shell/shell-pregel-start-cluster.js [hunk 2]
```diff
@@ -134,30 +135,21 @@ function basicTestSuite() {
...
-    if (i === 0) {
-      assertTrue(false, "timeout in pregel execution");
-    }
+    db[vColl].all().toArray()
+      .forEach(function (d) {
+        if (d[a] && d[a] !== -1) {
+          var diff = Math.abs(d[a] - d.result);
+          if (diff > EPS) {
+            console.log("Error on " + JSON.stringify(d));
+            assertTrue(false);// peng
```
Comment (markuspf) [inline tests/js/common/shell/shell-pregel-start-cluster.js:149]: I think
```suggestion
            assertTrue(false, "Error on " + JSON.stringify(d));// peng
```
might be more helpful as failure output?

File: tests/js/common/shell/shell-pregel-start-cluster.js [hunk 3]
```diff
@@ -251,60 +243,50 @@ function basicTestSuite() {
 
     // test the PREGEL_RESULT AQL function
     testPageRankAQLResult: function () {
-      var pid = pregel.start("pagerank", graphName, { shardKeyAttribute: shardKey, threshold: EPS / 10, store: false });
-      var i = 10000;
-      do {
-        internal.wait(0.2);
```
Comment (markuspf) [inline tests/js/common/shell/shell-pregel-start-cluster.js:257]: :partying_face:

**Comment by dothebart** [comment]
Hi, 
will this as well address the unstabilities that we see in the coverage runs with `shell_pregel_random.js` ?
```
    [FAILED]  tests/js/common/shell/shell-pregel-random.js

      "testAllJobsStatus" failed: Error: at assertion #15036: timeout in pregel execution: (false) does not evaluate to true
(Error
    at assertTrue (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:71:19)
    at Object.testAllJobsStatus (tests/js/common/shell/shell-pregel-random.js:312:7)
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:552:27)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:299:24)
    at tests/js/common/shell/shell-pregel-random.js:319:9
    at tests/js/common/shell/shell-pregel-random.js:321:3
```

**Comment by markuspf** [comment]
> Hi, will this as well address the unstabilities that we see in the coverage runs with `shell_pregel_random.js` ?

It might; the failure looks to be of the same type. At the latest we will know once this PR is merged, I guess.

**Comment by jvolmer** [comment]
@dothebart No, I haven't changed this test you mentioned. If it fails, the garbage collection did not work as expected. Just now, I improved the code of this test and its error message, maybe this helps the next time this test fails.

File: tests/js/common/shell/shell-pregel-components.js (hunk 1 above)
Comment (jvolmer) [reply tests/js/common/shell/shell-pregel-components.js]: Not sure if I found now all occurences: I found 4 and replaced them with a function.

**Comment by jvolmer** [comment]
@dothebart As already mentioned, I  found a wrong assumption in the testAllJobsStatus test and think I fixed it now.

**Pull request merged**